runtime/src/main/scala/tailcall/runtime/model/UnsafeSteps.scala

```diff
@@ -92,6 +95,13 @@ object UnsafeSteps {
     }
 
     object Http {
+      implicit val urlCodec: JsonCodec[URL] = JsonCodec[String].transformOrFail[URL](
+        string =>
+          try Right(URI.create(string).toURL)
+          catch { case _: Throwable => Left(s"Malformed url: ${string}") },
+        _.toString,
+      )
```

Comment:
Find a common place to put this duplicate logic.

Can also create a PR in ZIO-JSON for JsonCodec[URL], if it doesn't exist.

Reply:
We can add an `object JsonCodecImplicits` inside of `tailcall.runtime.internal`.
This would contain the implicit defs and then we could import it wherever required.